source/numerics/data_out_stack.cc

```diff
@@ -112,23 +112,16 @@ DataOutStack<dim, spacedim, DoFHandlerType>::declare_data_vector(
 
   // also make sure that no name is
   // used twice
-  for (std::vector<std::string>::const_iterator name = names.begin();
-       name != names.end();
-       ++name)
+  for (const auto &name : names)
     {
-      for (typename std::vector<DataVector>::const_iterator data_set =
-             dof_data.begin();
-           data_set != dof_data.end();
-           ++data_set)
-        for (unsigned int i = 0; i < data_set->names.size(); ++i)
-          Assert(*name != data_set->names[i], ExcNameAlreadyUsed(*name));
-
-      for (typename std::vector<DataVector>::const_iterator data_set =
-             cell_data.begin();
-           data_set != cell_data.end();
-           ++data_set)
-        for (unsigned int i = 0; i < data_set->names.size(); ++i)
-          Assert(*name != data_set->names[i], ExcNameAlreadyUsed(*name));
+      (void)name;
+      for (const auto &data_set : dof_data)
+        for (const auto &data_set_name : data_set.names)
+          Assert(name != data_set_name, ExcNameAlreadyUsed(name));
+
+      for (const auto &data_set : cell_data)
+        for (const auto &data_set_name : data_set.names)
+          Assert(name != data_set_name, ExcNameAlreadyUsed(name));
     };
```

Comment:
I think `git` is complaining about a merge conflict due to the semicolon I removed in #7600.

Reply:
Resolved.